subsys/dfu/dfu_target/src/dfu_target_mcuboot.c

```diff
@@ -185,13 +189,31 @@ int dfu_target_mcuboot_done(bool successful)
 	}
 
 	if (successful) {
+#ifdef CONFIG_FLASH_HAS_EXPLICIT_ERASE
+		stream_flash_erase_flatten_t erase_function = stream_flash_erase_page;
+#else
+		stream_flash_erase_flatten_t erase_function = NULL;
+#endif
+#ifdef CONFIG_FLASH_HAS_NO_EXPLICIT_ERASE
+		const struct flash_parameters *fparams = flash_get_parameters(secondary_dev[curr_sec_img]);
```

Comment:
runtime query is only needed when: #if defined(CONFIG_FLASH_HAS_EXPLICIT_ERASE) && defined (CONFIG_FLASH_HAS_NO_EXPLICIT_ERASE) is the case.

Reply:
Yes, I understand that and that was my intention when introducing them, but they were supposed to be translated into module specific Kconfig options; nevertheless they have no use here any way, because the flash_flatten internally does what you want, exactly the same way.
So all you have to do is just call `flash_flatten` on area you intend to remove data from and that will happen a proper, depending on a device.